build.rs

```diff
@@ -74,27 +81,57 @@ impl Registry {
 					account_type
 				))
 			}
-			if let Some(sig_type) = &account_type.standard_account {
-				match sig_type.as_str() {
-                    "Sr25519"| "Ed25519" | "secp256k1" | "*25519" => {},
-                    _ => {
-                        return Err(format!("Unknown sig type in standardAccount: expected one of Sr25519, Ed25519, secp256k1, *25519: {:?}", account_type))
-                    }
-                }
+			for (name, decimals) in account_type.symbols.iter().zip(&account_type.decimals) {
+				use Entry::*;
+				let network = account_type.display_name.clone();
+				match tokens.entry(name.to_owned()) {
+					Occupied(mut e) => {
+						if e.get().decimals != *decimals {
+							return Err(format!(
+								"Inconsistent decimals specified for token {}.\nPrevious networks: {}\nCurrent network: {}",
+								name,
+								e.get().networks.join(", "),
+								network,
+							))
+						}
+						e.get_mut().networks.push(network);
+					},
+					Vacant(e) => {
+						e.insert(TokenType {
+							symbol: name.to_owned(),
+							decimals: *decimals,
+							networks: vec![network],
+						});
+					},
+				}
 			}
 		}
-		Ok(())
+		let mut x = tokens.into_iter().map(|(_k, v)| v).collect::<Vec<_>>();
+		x.sort_by_key(|i| i.symbol.to_owned());
+		Ok((self.accounts, x))
```

Comment:
```suggestion
		Ok((self.accounts, tokens.into_values().collect()))
```

If you use `BTreeMap` then you don't need any of the other code.